modules/rosa-sts-cluster-terraform.adoc

```diff
@@ -0,0 +1,532 @@
+  source  = "terraform-redhat/rosa-sts/aws"
+  version = "0.0.15"
+
+  create_operator_roles = false
+  create_oidc_provider  = true
+
+  cluster_id                  = ""
+  rh_oidc_provider_thumbprint = rhcs_rosa_oidc_config.oidc_config.thumbprint
+  rh_oidc_provider_url        = rhcs_rosa_oidc_config.oidc_config.oidc_endpoint_url
+  tags                        = var.additional_tags
+  path                        = var.path
+}
+EOF
+----
+
+.. Run the following command to create your `operator-roles.tf` file:
++
+[source,terminal]
+----
+$ cat<<-EOF>operator-roles.tf
+module "operator_roles" {
+  source  = "terraform-redhat/rosa-sts/aws"
+  version = "0.0.15"
+
+  create_operator_roles = true
+  create_oidc_provider  = false
+
+  rh_oidc_provider_thumbprint = rhcs_rosa_oidc_config.oidc_config.thumbprint
+  rh_oidc_provider_url        = rhcs_rosa_oidc_config.oidc_config.oidc_endpoint_url
+  operator_roles_properties   = data.rhcs_rosa_operator_roles.operator_roles.operator_iam_roles
+  tags                        = var.additional_tags
+  path                        = var.path
+}
+EOF
+----
+
+.. Run the following command to create your `variables.tf` file:
++
+[source,terminal]
+----
+$ cat<<-EOF>variables.tf
+variable "redhat_aws_account_id" {
+  type    = string
+}
+
+variable "ocm_environment" {
+  type    = string
+  default = "production"
+}
+
+variable "rosa_openshift_version" {
+  type        = string
+  default     = "4.13"
+  description = "Desired version of OpenShift for the cluster, for example '4.1.0'. If version is greater than the currently running version, an upgrade will be scheduled."
+}
+
+# Account Roles
+  variable "account_role_prefix" {
+  type    = string
+  default = "ROSA"
+}
+
+variable "account_role_policies" {
+  description = "account role policies details for account roles creation"
+  type = object({
+    sts_installer_permission_policy             = string
+    sts_support_permission_policy               = string
+    sts_instance_worker_permission_policy       = string
+    sts_instance_controlplane_permission_policy = string
+  })
+  default = null
+}
+
+# Operator Roles
+  variable "operator_role_prefix" {
+  type    = string
+  default = "ROSA"
+}
+
+variable "operator_role_policies" {
+  description = "operator role policies details for operator roles creation"
+  type = object({
+    openshift_cloud_credential_operator_cloud_credential_operator_iam_ro_creds_policy = string
+    openshift_cloud_network_config_controller_cloud_credentials_policy                = string
+    openshift_cluster_csi_drivers_ebs_cloud_credentials_policy                        = string
+    openshift_image_registry_installer_cloud_credentials_policy                       = string
+    openshift_ingress_operator_cloud_credentials_policy                               = string
+    openshift_machine_api_aws_cloud_credentials_policy                                = string
+  })
+  default = null
+}
+
+variable "managed_oidc" {
```

Comment:
As this example forces a managed OIDC provider, this variable can be removed for the sake of readability